jobs/asset-sender/src/main/java/com/paladincloud/common/assets/Assets.java

```diff
@@ -0,0 +1,473 @@
+package com.paladincloud.common.assets;
+
+import static java.util.Map.entry;
+
+import com.paladincloud.common.DocumentFields;
+import com.paladincloud.common.aws.Database;
+import com.paladincloud.common.aws.S3;
+import com.paladincloud.common.config.ConfigConstants;
+import com.paladincloud.common.config.ConfigConstants.Sender;
+import com.paladincloud.common.config.ConfigService;
+import com.paladincloud.common.config.Types;
+import com.paladincloud.common.errors.JobException;
+import com.paladincloud.common.search.ElasticBatch;
+import com.paladincloud.common.search.ElasticBatch.BatchItem;
+import com.paladincloud.common.search.ElasticSearch;
+import com.paladincloud.common.util.MapExtras;
+import com.paladincloud.common.util.StringExtras;
+import com.paladincloud.common.util.TimeFormatter;
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Assets {
+
+    private static final Logger LOGGER = LogManager.getLogger(Assets.class);
+    private static final String OVERRIDE_PREFIX = "pac_override_";
+    private static final String TYPE_ON_PREM_SERVER = "onpremserver";
+
+    public static void upload(String dataSource) {
+        var bucket = ConfigService.get(ConfigConstants.S3.BUCKET_NAME);
+        var allFilenames = S3.listObjects(ConfigService.get(ConfigConstants.S3.BUCKET_NAME),
+            assetsPathPrefix(dataSource));
+        var types = Types.getTypesWithDisplayName(dataSource);
+        // TODO: DON'T CHECK THIS DEV HACK IN!
```

Comment:
Yes, I've pushed incomplete code; this is to prevent me from merging this without completing it.
The dev hack will change; I intend to add an optional command line parameter or two to help with this